modules/widgets/blog-stats.php

```diff
@@ -0,0 +1,177 @@
+		);
+	}
+
+	/**
+	 * Return All Time Stats for that blog.
+	 *
+	 * We query the WordPress.com Stats REST API endpoint.
+	 *
+	 * @uses stats_get_from_restapi(). That function caches data locally for 5 minutes.
+	 *
+	 * @return string|false $views All Time Stats for that blog.
+	 */
+	public function get_stats() {
+		// Get data from the WordPress.com Stats REST API endpoint.
+		$stats = stats_get_from_restapi( array( 'fields' => 'stats' ) );
+
+		if (
+			isset( $stats )
+			&& ! empty( $stats )
+			&& isset( $stats->stats->views )
+		) {
+			return $stats->stats->views;
+		} else {
+			return false;
+		}
+	}
+
+	/**
+	 * Back-end widget form.
+	 *
+	 * @see WP_Widget::form()
+	 *
+	 * @param array $instance Previously saved values from database.
+	 *
+	 * @return void
+	 */
+	function form( $instance ) {
+		$instance = wp_parse_args( $instance, $this->defaults() );
+		?>
+
+		<p>
+			<label for="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>"><?php esc_html_e( 'Title:', 'jetpack' ); ?></label>
+			<input class="widefat" id="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'title' ) ); ?>" type="text" value="<?php echo esc_attr( $instance['title'] ); ?>" />
+		</p>
+		<p>
+			<label for="<?php echo esc_attr( $this->get_field_id( 'hits' ) ); ?>"><?php echo number_format_i18n( '12345' ); ?></label>
+			<input class="widefat" id="<?php echo esc_attr( $this->get_field_id( 'hits' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'hits' ) ); ?>" type="text" value="<?php echo esc_attr( $instance['hits'] ); ?>" />
+		</p>
+		<p><?php esc_html_e( 'Hit counter is delayed by up to 60 seconds.', 'jetpack' ); ?></p>
+
+		<?php
+	}
+
+	/**
+	 * Sanitize widget form values as they are saved.
+	 *
+	 * @see WP_Widget::update()
+	 *
+	 * @param array $new_instance Values just sent to be saved.
+	 * @param array $old_instance Previously saved values from database.
+	 *
+	 * @return array Updated safe values to be saved.
+	 */
+	function update( $new_instance, $old_instance ) {
+		$instance          = array();
+		$instance['title'] = wp_kses( $new_instance['title'], array() );
+		$instance['hits']  = wp_kses( $new_instance['hits'], array() );
+
+		return $instance;
+	}
+
+	/**
+	 * Front-end display of widget.
+	 *
+	 * @see WP_Widget::widget()
+	 *
+	 * @param array $args     Widget arguments.
+	 * @param array $instance Saved values from database.
+	 */
+	function widget( $args, $instance ) {
+		$instance = wp_parse_args( $instance, $this->defaults() );
+
+		/** This filter is documented in wp-includes/widgets/class-wp-widget-pages.php */
+		$title = apply_filters( 'widget_title', $instance['title'] );
+
+		echo $args['before_widget'];
+
+		if ( ! empty( $title ) ) {
+			echo $args['before_title'] . esc_html( $title ) . $args['after_title'];
+		}
+
+		// Get the Site Stats.
+		$views = $this->get_stats();
+
+		if ( isset( $views ) && ! empty( $views ) ) {
```

Comment:
Based on `get_stats` return, `views` will always be set to something, so it's not necessary to check `isset`.